File: ios/BundleReloader.m
//
//  BundleReloader.m
//  OTAApp
//
//  Created by Aakash G S on 19/05/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BundleReloader, NSObject)

RCT_EXTERN_METHOD(reloadBridge)

@end

File: ios/BundleReloader.swift
@objc(BundleReloader)
class BundleReloader: NSObject {
  @objc static func reloadBridge() {
    DispatchQueue.main.async {
      guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
            let window = appDelegate.window,
            let factory = appDelegate.reactNativeFactory else {
        print("❌ Could not access AppDelegate or window")
        return
      }

      print("🔁 Restarting React Native with updated JS bundle")
      factory.startReactNative(
        withModuleName: "DemoApp",
        in: window,
        launchOptions: nil
      )
    }
  }

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

